### PR TITLE
#3615 - Partner current year income path - Part 1

### PR DESCRIPTION
**Pull request by dheepak-aot**
# Student Application - Partner current year income

## Student application form update

- [x] The form input `IsYourPartnerAbleToReport` which was coming from BCSC question is now converted to a form.io calculated variable which returns `true | false` instead of `"yes" | "no"`.

- [x] Added a question to enable the current year partner income submission in the partner information panel. The question blurb consists of the `dynamic current year`

image

- [x] The info and warning banners are delayed until both BCSC and Current year income questions are answered.

image

- [x] When the question for current year income is answered as `Yes` show the current year income question with options and file upload instead of current tax return income. 

image

- [x] When student enters the current year income for the partner instead of the current tax return income, then choose the self declaration path with exception instead of supporting user input even if the partner is eligible for BC Services Card.

image

image

image

image

image

## Content Updates

- [x] The content (header and the blurb) updated for the previously called `Partner Income` section.
image

## Assessment gateway updates

- [x] Minor update to `assessment-gateway` and `fulltime-assessment-*` to use the input `IsYourPartnerAbleToReport` to evaluate `true | false` instead of `"yes" | "no"`

## Workflow E2E test updates
- [x] Changed the data type of `studentDataIsYourPartnerAbleToReport` to boolean from YesNo.

## Upcoming PR

- `part-time-assessment-*` changes
- Workflow E2E tests.

### CODE REVIEW

File: sources/packages/backend/workflow/test/2022-2023/parttime-assessment/awards-amounts/parttime-assessment-awards-amount-BCAG.e2e-spec.ts [hunk 1]
```diff
@@ -165,8 +160,7 @@ describe(`E2E Test Workflow parttime-assessment-${PROGRAM_YEAR}-awards-amount-BC
       createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
     assessmentConsolidatedData.studentDataCRAReportedIncome = 20001;
     assessmentConsolidatedData.studentDataRelationshipStatus = "married";
-    assessmentConsolidatedData.studentDataIsYourPartnerAbleToReport =
-      YesNoOptions.Yes;
+    assessmentConsolidatedData.studentDataIsYourPartnerAbleToReport = true;
```
[the remark this answers is not in the document]
Comment (dheepak-aot) [reply sources/packages/backend/workflow/test/2022-2023/parttime-assessment/awards-amounts/parttime-assessment-awards-amount-BCAG.e2e-spec.ts:163]: These are existing E2E tests based on studentDataIsYourPartnerAbleToReport = YesNoOptions.Yes;

I do not see a scenario where it is set to No(false).

File: sources/packages/forms/src/form-definitions/studentexceptions.json [hunk 2]
```diff
@@ -1947,8 +1944,49 @@
...
+          "lockKey": true
+        },
+        {
+          "clearOnHide": false,
+          "key": "currentYearPartnerIncomeApplicationException",
+          "input": false,
+          "components": [
+            {
+              "key": "currentYearPartnerIncomeApplicationExceptionCurrentYearPartnerIncome",
+              "label": "Current Year Partner Income",
```
Comment (andrewsignori-aot) [inline sources/packages/forms/src/form-definitions/studentexceptions.json:1956]: Only the last two exceptions in this list seem to be using a different case.
This one should be `Current year partner's income` probably.
Not a blocker if it is following business requirements.

image

**Review by andrewsignori-aot — APPROVED** [review]
Great work, looks good 👍

File: sources/packages/forms/src/form-definitions/studentexceptions.json (hunk 2 above)
Comment (dheepak-aot) [reply sources/packages/forms/src/form-definitions/studentexceptions.json:1956]: I missed the content from AC. Updated it. 
image

**Pull request merged**